[2025-06-27T04:33:26.089+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-06-27T04:33:26.133+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_processing_pipeline.Structure_Data_With_Spark manual__2025-06-27T04:33:19.078517+00:00 [queued]>
[2025-06-27T04:33:26.140+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_processing_pipeline.Structure_Data_With_Spark manual__2025-06-27T04:33:19.078517+00:00 [queued]>
[2025-06-27T04:33:26.141+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-06-27T04:33:26.282+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): Structure_Data_With_Spark> on 2025-06-27 04:33:19.078517+00:00
[2025-06-27T04:33:26.288+0000] {standard_task_runner.py:64} INFO - Started process 215 to run task
[2025-06-27T04:33:26.292+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spark_processing_pipeline', 'Structure_Data_With_Spark', 'manual__2025-06-27T04:33:19.078517+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmpsssrg2v1']
[2025-06-27T04:33:26.295+0000] {standard_task_runner.py:91} INFO - Job 8: Subtask Structure_Data_With_Spark
[2025-06-27T04:33:26.732+0000] {task_command.py:426} INFO - Running <TaskInstance: spark_processing_pipeline.Structure_Data_With_Spark manual__2025-06-27T04:33:19.078517+00:00 [running]> on host 3b2681f5d151
[2025-06-27T04:33:27.447+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_processing_pipeline' AIRFLOW_CTX_TASK_ID='Structure_Data_With_Spark' AIRFLOW_CTX_EXECUTION_DATE='2025-06-27T04:33:19.078517+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-27T04:33:19.078517+00:00'
[2025-06-27T04:33:27.449+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-06-27T04:33:27.487+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-06-27T04:33:27.489+0000] {spark_submit.py:401} INFO - Spark-Submit cmd: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --queue root.default /opt/***/include/scripts/structured_data.py /opt/***/data/raw/raw.json /opt/***/data/foundation
[2025-06-27T04:33:27.570+0000] {spark_submit.py:571} INFO - /home/***/.local/lib/python3.10/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-06-27T04:33:27.571+0000] {spark_submit.py:571} INFO - JAVA_HOME is not set
[2025-06-27T04:33:27.572+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-06-27T04:33:27.606+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 502, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --queue root.default /opt/***/include/scripts/structured_data.py /opt/***/data/raw/raw.json /opt/***/data/foundation. Error code is: 1.
[2025-06-27T04:33:27.610+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_processing_pipeline, task_id=Structure_Data_With_Spark, run_id=manual__2025-06-27T04:33:19.078517+00:00, execution_date=20250627T043319, start_date=20250627T043326, end_date=20250627T043327
[2025-06-27T04:33:27.657+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 8 for task Structure_Data_With_Spark (Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --queue root.default /opt/***/include/scripts/structured_data.py /opt/***/data/raw/raw.json /opt/***/data/foundation. Error code is: 1.; 215)
[2025-06-27T04:33:27.669+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2025-06-27T04:33:27.716+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-06-27T04:33:27.718+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
